app/controllers/application_controller.rb

```diff
@@ -125,4 +125,10 @@ def set_current_context_for_logging!
   rescue => e
     Rails.logger.error(e)
   end
+
+  def instrument_with_timer
+    @timer = Timer.new("#{params[:controller]}.#{params[:action]}")
```

Comment:
any chance that `@timer` could be created twice, overwriting the first instance?

Reply:
I believe each action is called only once. Are you thinking of any specific scenario?
We use this pattern to fill in other parameters like `@sample`...

Reply:
No, it just requires the programmer to understand that. Maybe add a comment "This should only be called once per request" or put 

`raise "must be called only once" unless @timer.nil?`

Reply:
It is not "must" but a "should"... thus added logging.
Do not want instrumentation to be a blocker and thus prefer not raising exception, since the application will not get into an error state.